package/cloudshell/iac/terraform/tagging/tag_terraform_resources.py

```diff
@@ -642,12 +646,16 @@ def start_tagging_terraform_resources(main_dir_path: str, logger, tags_dict: dic
                                          f" (Return_code is {return_code})"
                                          f"\n\nErrors are:\n{stderr}\n")
                 LoggerHelper.write_error("Tagging terraform resources operation has FAILED !!!!!")
-                exit(1)
+                # Had to change exit(1) to "raise" so exception can be handled outside
+                # exit(1)
+                raise ValueError("Errors were found in the last validation check")
 
         else:
             LoggerHelper.write_warning("No untaggable resources were found, but errors in plan file do exists")
             LoggerHelper.write_error("Tagging terraform resources operation has FAILED !!!!!")
-            exit(1)
+            # Had to change exit(1) to "raise" so exception can be handled outside
+            # exit(1)
+            raise ValueError("No untaggable resources were found, but errors in plan file do exists")
```

Comment:
Suggested error message:
"No untaggable resources were found, but there is an error in Terraform Plan when applying automated tags. Please check the logs for more details"